torch/utils/data/dataloader.py

```diff
@@ -116,15 +125,23 @@ class DataLoader(object):
               details on these two types of datasets and how
               :class:`~torch.utils.data.IterableDataset` interacts with `Multi-process data loading`_.
     """
+    dataset: Dataset[T_co]
+    batch_size: Optional[int]
+    num_workers: int
+    pin_memory: bool
+    drop_last: bool
+    timeout: float
 
     __initialized = False
 
-    def __init__(self, dataset, batch_size=1, shuffle=False, sampler=None,
-                 batch_sampler=None, num_workers=0, collate_fn=None,
-                 pin_memory=False, drop_last=False, timeout=0,
-                 worker_init_fn=None, multiprocessing_context=None,
-                 generator=None):
-        torch._C._log_api_usage_once("python.data_loader")
+    def __init__(self, dataset: Dataset[T_co], batch_size: Optional[int] = 1,
```

Comment:
`batch_size` was original `int`, why did this have to change?

Reply:
There is a condition in the body of this method which checks whether `batch_size` is `None`, so `None` is an allowed value.

Separately from this, this argument is also used as a local variable in the method and sometimes `None` is assigned to it.